sklearn/tree/_classes.py

```diff
@@ -1185,6 +1320,9 @@ class DecisionTreeRegressor(RegressorMixin, BaseDecisionTree):
         :ref:`sphx_glr_auto_examples_tree_plot_unveil_tree_structure.py`
         for basic usage of these attributes.
 
+    builder_ : TreeBuilder instance
+        The underlying TreeBuilder object.
+
```

Comment:
Is there any state preserved in the initial builder that we want to preserve when we call update? 

I.e. do we need to preserve the builder necessarily?

Reply:
I need to check but maybe not.

Reply:
The `builder` object is normally initialized in `fit` and contains some parameters like `min_samples_split`, `splitter` and `criterion`. The initialization (checks and stuff) of these parameters include many lines of code, so I'll just leave the object as it is for now. We can optimize it later by modulating those steps.